src/HasRole.php

```diff
@@ -23,13 +23,17 @@ public function getRole(): Role
 
     public function hasRole($role): bool
     {
-        if (is_int($role)) {
-            return $this->role_id == $role;
-        } else if (is_string($role)) {
-            return strcasecmp($this->role->slug, $role) == 0;
-        } else {
-            throw new NotImplementedException("function hasRole() accepts either int (role_id) or string (role name). "
-                .gettype($role)." was given");
+        if ($this->role) {
```

Comment:
This way relation will be loaded even if check could be made by ID. Ex.

```php
$user->hasRole(5)
```

Technically, we can compare IDs without additional SQL request. But ```if ($this->role)``` check will cause relation to be loaded before IDs comparison.

Reply:
theoretically, the "role" field can be calculated, but in this case, with the current implementation, the check by the role ID may not work properly.